vega_sim/network_service.py

```diff
@@ -337,6 +342,83 @@ def core_client(self) -> None:
         )
         pass
 
+    def check_datanode(self, raise_on_error: Optional[bool] = True):
+        """Checks if the current data-node connection is healthy.
+
+        If the current data-node connection has timed out or the end-point is
+        unresponsive then the service will attempt to establish a connection
+        with a new healthy data-node.
+
+        Args:
+            max_attempts (int, optional):
+                Maximum number of connection attempts to attempt before raising an
+                error. Defaults to -1 (infinite attempts).
+        """
+
+        try:
+            self.ping_datanode()
+            return
+
+        except grpc.FutureTimeoutError:
+            # Unable to connect to datanode
+            msg = f"Connection to endpoint {self._data_node_grpc_url} timed out."
+            if raise_on_error:
+                raise RuntimeError(msg)
```

Comment:
Same as above, raise the actual error thrown (Although we're at least raising a specific error here)